### PR TITLE
Account unification - avatar selection

### PR DESCRIPTION
**Pull request by bobo-k2**
**Pull Request Summary**

This PR implements an option to select a NFT as unified account icon
image
image

Send me a message with your evm address do I can mint some test NFTs for your

**Check list**

- [ ] contains breaking changes
- [x] adds new feature
- [x] modifies existing feature (bug fix or improvements)
- [ ] relies on other tasks
- [ ] documentation changes
- [ ] tested on mobile devices

### CODE REVIEW

File: src/components/header/modals/account-unification/SelectNft.vue [hunk 1]
```diff
@@ -2,46 +2,73 @@
...
         class="item"
-        :class="nft.isSelected && 'item--selected'"
+        :class="selectedIndex === index && 'item--selected'"
+        @click="selectedIndex = index"
       >
-        <img :src="nft.img" alt="NFT" />
+        <img :src="nft.image" :alt="nft.name" />
       </div>
     </div>
+    <div v-if="ownedNfts.length === 0 && !isBusy" class="item">
```
Comment (impelcrypto) [inline src/components/header/modals/account-unification/SelectNft.vue:14]: Let's use i18

File: src/components/header/modals/ModalAccountUnification.vue [hunk 2]
```diff
@@ -61,13 +67,21 @@
...
           :selected-evm-address="selectedEvmAddress"
+          :avatar-url="avatar?.image ?? ''"
           :is-busy="isLoading"
           @next="updateSteps(6)"
         />
       </div>
       <div v-else-if="currentStep === 6">
         <au-step6 />
       </div>
+      <div v-else-if="currentStep === 200">
```
Comment (impelcrypto) [inline src/components/header/modals/ModalAccountUnification.vue:78]: Perhaps we might want to make a enum for the number of `currentStep` or make some comments.

**Review by impelcrypto — CHANGES_REQUESTED** [review]

**Comment by bobo-k2** [comment]
> 0x4184C9f4dF0Ab75F3A8ab337Fc07BDF7b17a7918

Sent

File: src/components/header/modals/ModalAccountUnification.vue (hunk 2 above)
Comment (bobo-k2) [reply src/components/header/modals/ModalAccountUnification.vue:78]: > Perhaps we might want to make a enum for the number of `currentStep` or make some comments.

I was thinking about that, but too lazy to implement

Comment (bobo-k2) [reply src/components/header/modals/ModalAccountUnification.vue]: Fixed in https://github.com/AstarNetwork/astar-apps/pull/975/commits/63a6481e3ee117d41c5c1b5cd1679a4d8b1d34a0

**Comment by impelcrypto** [comment]
Perhaps display some words (like 'loading...') while fetching the NFTs?
image

**Comment by impelcrypto** [comment]
Let's display something like 'There are no NFTs found' if the account doesn't have NFTs.

image

**Comment by bobo-k2** [comment]
> Let's display something like 'There are no NFTs found' if the account doesn't have NFTs.

There is a message if no NFTs. In case above there is something found, but it looks wrong. Please share details how to reproduce the error (evm address)

**Comment by bobo-k2** [comment]
> Perhaps display some words (like 'loading...') while fetching the NFTs? 

Fetching NFTs metadata from API is fast. Loading images into browser can take some time, because NFT images are quite big. Since this is under a browser control I will see if I can do something. There should be some events on `img` tag

**Comment by impelcrypto** [comment]
> There is a message if no NFTs. In case above there is something found, but it looks wrong. Please share details how to reproduce the error (evm address)

hmm... I can't reproduce the issue again.  

Yes, this is good!
image

**Review by impelcrypto — APPROVED** [review]

**Comment by gluneau** [comment]
Editing an existing Unified account when there is no NFT chosen yet.
image

**Review by gluneau — APPROVED** [review]
LGTM except for the comment about the NFT picture not showing up on edit.

**Comment by bobo-k2** [comment]
> LGTM except for the comment about the NFT picture not showing up on edit.

Thanks for the review. Fixed the issue in https://github.com/AstarNetwork/astar-apps/pull/975/commits/c4cb432f810c31d1109a6ac3dae490585e4f4829

**Pull request merged**